tools_docker/Debian_New_Uwsgi/common.sh

```diff
@@ -0,0 +1,235 @@
+# Create log folder for Redis (necessary when migrate from Tracim < 3.0.0 )
+if [ ! -d /var/tracim/logs/pushpin ]; then
+    mkdir /var/tracim/logs/pushpin -p
+    touch /var/tracim/logs/pushpin/access_7999.log
+    touch /var/tracim/logs/pushpin/error_7999.log
+    touch /var/tracim/logs/pushpin/m2adapter.log
+    touch /var/tracim/logs/pushpin/mongrel2_7999.log
+    touch /var/tracim/logs/pushpin/pushpin-handler.log
+    touch /var/tracim/logs/pushpin/pushpin-proxy.log
+    chown www-data:www-data -R /var/tracim/logs/pushpin
+    chmod 775 -R /var/tracim/logs/pushpin
+fi
+# Create Zurl log (necessary when migrate from Tracim < 3.0.0 )
+if [ ! -f /var/tracim/logs/zurl.log ];then
+    touch /var/tracim/logs/zurl.log
+    chown www-data:www-data /var/tracim/logs/zurl.log
+    chmod 775 /var/tracim/logs/zurl.log
+fi
+
+# Create symbollic link to easy find log in container folder
+# create_symlink_if_not_exist
+create_log_symlink /var/tracim/logs/tracim_web.log /var/log/uwsgi/app/tracim_web.log
+create_log_symlink /var/tracim/logs/tracim_webdav.log /var/log/uwsgi/app/tracim_webdav.log
+create_log_symlink /var/tracim/logs/tracim_caldav.log /var/log/uwsgi/app/tracim_caldav.log
+create_log_symlink /var/tracim/logs/apache2-access.log /var/log/apache2/tracim-access.log
+create_log_symlink /var/tracim/logs/apache2-error.log /var/log/apache2/tracim-error.log
+create_log_symlink /var/tracim/logs/redis/redis-server.log /var/log/redis-server.log
+create_log_symlink var/tracim/logs/pushpin/access_7999.log /var/log/access_7999.log
+create_log_symlink /var/tracim/logs/pushpin/error_7999.log /var/log/error_7999.log
+create_log_symlink /var/tracim/logs/pushpin/m2adapter.log /var/log/m2adapter.log
+create_log_symlink /var/tracim/logs/pushpin/mongrel2_7999.log /var/log/mongrel2_7999.log
+create_log_symlink /var/tracim/logs/pushpin/pushpin-handler.log /var/log/pushpin-handler.log
+create_log_symlink /var/tracim/logs/pushpin/pushpin-proxy.log /var/log/pushpin-proxy.log
+create_log_symlink /var/tracim/logs/zurl.log /var/log/zurl.log
+
+# Modify default log path for Pushpin, Redis, Zurl (since Tracim 3.0.0)
+sed -i "s|^logdir=.*|logdir=/var/tracim/logs/pushpin/|g" /etc/pushpin/pushpin.conf
+sed -i "s|^logfile.*|logfile /var/tracim/logs/redis/redis-server.log|g" /etc/redis/redis.conf
+sed -i "s|^DAEMON_ARGS=.*|DAEMON_ARGS=\"--config=/etc/zurl.conf --logfile=/var/tracim/logs/zurl.log\" # Arguments to run the daemon with|g" /etc/init.d/zurl
+
+# Add user Pushpin, Redis, Zurl in www-data group for logging (since Tracim 3.0.0)
+adduser redis www-data
+adduser pushpin www-data
+adduser zurl www-data
+
+# Create uWSGi app folder and set right
+if [ ! -d /var/run/uwsgi/app ]; then
+    mkdir /var/run/uwsgi/app -p
+    chown www-data:www-data -R /var/run/uwsgi
+fi
+
+# Create Tracim required folder
+create_dir /var/tracim/data
+create_dir /var/tracim/assets
+create_dir /var/tracim/data/sessions_data
+create_dir /var/tracim/data/sessions_lock
+create_dir /var/tracim/data/depot
```

Comment:
not necessary but i did this because i wanted to change right here at the same time as i create dir, finally don't need it but it's more easy to read, that many redundant if.

Reply:
I suggest we clean those things up in January. We'll need to get back to this PR when doing this.

Reply:
yes in #3968